tests/tlsio_esp8266_ut/tlsio_esp8266_ut.c

```diff
@@ -0,0 +1,1028 @@
+#endif
+
+void* my_gballoc_malloc(size_t size)
+{
+    return malloc(size);
+}
+
+void* my_gballoc_realloc(void* ptr, size_t size)
+{
+    return realloc(ptr, size);
+}
+
+void my_gballoc_free(void* ptr)
+{
+    free(ptr);
+}
+
+/**
+ * Include the C standards here.
+ */
+#include <stddef.h>
+#include <time.h>
+
+/**
+ * Include the test tools.
+ */
+#include "testrunnerswitcher.h"
+#include "umock_c.h"
+#include "umocktypes_charptr.h"
+#include "umock_c_negative_tests.h"
+#include "azure_c_shared_utility/macro_utils.h"
+
+/**
+ * Include the mockable headers here.
+ */
+#define ENABLE_MOCKS
+#include "../../adapters/esp8266_mock.h"
+#undef ENABLE_MOCKS
+
+/**
+ * Include the target header after the ENABLE_MOCKS session.
+ */
+#include "azure_c_shared_utility/xio.h"
+#include "azure_c_shared_utility/tlsio.h"
+
+static int g_ssl_write_success = 1;
+static int g_ssl_read_returns_data = 1;
+static int g_on_bytes_received_buffer_size = 0;
+static int g_ssl_set_fragment_success = 1;
+static int g_ssl_new_success = 1;
+static int g_ssl_set_fd_success = 1;
+static int g_ssl_connect_success = 1;
+
+#define MAX_RETRY_WRITE 500
+#define MAX_RETRY 10
+
+typedef enum TLSIO_STATE_TAG
+{
+    TLSIO_STATE_NOT_OPEN,
+    TLSIO_STATE_OPENING,
+    TLSIO_STATE_OPEN,
+    TLSIO_STATE_CLOSING,
+    TLSIO_STATE_ERROR
+} TLSIO_STATE;
+
+typedef struct TLS_IO_INSTANCE_TAG
+{
+    ON_BYTES_RECEIVED on_bytes_received;
+    ON_IO_OPEN_COMPLETE on_io_open_complete;
+    ON_IO_CLOSE_COMPLETE on_io_close_complete;
+    ON_IO_ERROR on_io_error;
+    void* on_bytes_received_context;
+    void* on_io_open_complete_context;
+    void* on_io_close_complete_context;
+    void* on_io_error_context;
+    SSL* ssl;
+    SSL_CTX* ssl_context;
+    TLSIO_STATE tlsio_state;
+    char* hostname;
+    int port;
+    char* certificate;
+    const char* x509certificate;
+    const char* x509privatekey;
+} TLS_IO_INSTANCE;
+
+int real_mallocAndStrcpy_s(char** destination, const char* source);
+const IO_INTERFACE_DESCRIPTION* tlsio_openssl_get_interface_description(void);
+
+
+int my_lwip_select(int maxfdp1, fd_set *readset, fd_set *writeset, fd_set *exceptset,
+               struct timeval *timeout){
+    return 1;
+}
+
+int my_SSL_write(SSL *ssl, const void *buffer, int len){
+    if (g_ssl_write_success){
+        return len;
+    }else{
+        return 0;
+    }
+}
+
+void my_SSL_CTX_free(SSL_CTX *ctx){
+
+}
+
+void my_SSL_free(SSL *ssl){
+
+}
+
+int my_SSL_read(SSL *ssl, void *buffer, int len){
+    if (g_ssl_read_returns_data){
+        return len;
+    }else{
+        return 0;
+    }
+}
+
+int my_SSL_connect(SSL *ssl){
+    if (g_ssl_connect_success == 1){
+        return 0;
+    }else{
+        return -1;
+    }
+}
+
+int my_SSL_set_fd(SSL *ssl, int fd){
+    if (g_ssl_set_fd_success == 1){
+        return 1;
+    }else{
+        return 0;
+    }
+}
+
+SSL* my_SSL_new(SSL_CTX *ssl_ctx){
+    if (g_ssl_new_success == 1){
+         SSL* ssl = (SSL*)malloc(sizeof(SSL));
```

Comment:
SSL is defined as 
```c
typedef void SSL;
```
So `sizeof(SSL)` is not allowed. It is the same for SSL_CTX.